src/query/storage/m3/accounted_series_iter_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+package m3
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/m3db/m3/src/dbnode/encoding"
+	"github.com/m3db/m3/src/dbnode/ts"
+	"github.com/m3db/m3/src/query/test/seriesiter"
+	"github.com/m3db/m3/src/x/cost"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func newTestEnforcer(limit cost.Cost) *cost.Enforcer {
+	limitObj := cost.Limit{Threshold: limit, Enabled: true}
+	return cost.NewEnforcer(
+		cost.NewStaticLimitManager(cost.NewLimitManagerOptions().SetDefaultLimit(limitObj)),
+
+		cost.NewTracker(),
+		nil,
+	)
+}
+
+func newSimpleLimit(c cost.Cost) cost.Limit {
+	return cost.Limit{
+		Threshold: c,
+		Enabled:   true,
+	}
+}
+
+// copied from query/cost ; factor out if needed.
+func assertCurCost(t *testing.T, expectedCost cost.Cost, ef cost.EnforcerIF) {
+	actual, _ := ef.State()
+	assert.Equal(t, cost.Report{
+		Cost:  expectedCost,
+		Error: nil,
+	}, actual)
+}
+
+type accountedSeriesIterSetup struct {
+	Ctrl     *gomock.Controller
+	Enforcer *cost.Enforcer
+	Iter     *AccountedSeriesIter
+}
+
+func setupAccountedSeriesIter(t *testing.T, numValues int, limit cost.Cost) *accountedSeriesIterSetup {
+	ctrl := gomock.NewController(t)
+	enforcer := newTestEnforcer(limit)
+
+	mockWrappedIter := seriesiter.NewMockSeriesIterator(ctrl, seriesiter.NewMockValidTagGenerator(ctrl), numValues)
+	return &accountedSeriesIterSetup{
+		Ctrl:     ctrl,
+		Enforcer: enforcer,
+		Iter:     NewAccountedSeriesIter(mockWrappedIter, enforcer),
+	}
+}
+
+func TestAccountedSeriesIter_Next(t *testing.T) {
+	t.Run("adds to enforcer", func(t *testing.T) {
```

Comment:
Nothing really; I can switch to `TestAccountedSeriesIter_Next_AddsToEnforcer` if you like.

Reply:
I do kind of like the logical grouping of things though (and it can allow for shared setup that's specific to the component, though I'm not doing that here).